### PR TITLE
Fix: Empty enum never type suggested only if the feature is enabled

### PR DESCRIPTION
**Pull request by nahuakang**
This PR addresses [Issue 6422](https://github.com/rust-lang/rust-clippy/issues/6422). Instead of always recommending `never type` for empty enums, Clippy would only recommend [the lint](https://rust-lang.github.io/rust-clippy/master/index.html#empty_enum) if [LatePass.TyCtxt](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html) has `features().never_type` enabled.

- \[ ] Followed [lint naming conventions][lint_naming]
- \[x] Added passing UI tests (including committed `.stderr` file)
- \[x] `cargo test` passes locally
- \[x] Executed `cargo dev update_lints`
- \[x] Added lint documentation
- \[x] Run `cargo dev fmt`
---

*Please write a short comment explaining your change (or "none" for internal only changes)*
changelog: Only trigger [`empty_enum`] lint if `never_type` feature is enabled.

### CODE REVIEW

File: clippy_lints/src/empty_enum.rs [hunk 1]
```diff
@@ -8,6 +8,10 @@ use rustc_session::{declare_lint_pass, declare_tool_lint};
 declare_clippy_lint! {
     /// **What it does:** Checks for `enum`s with no variants.
     ///
+    /// As of this writing, the never type is still a
+    /// nightly-only experimental API. Therefore, this lint is only triggered
+    /// if the never type is enabled
```
Comment (flip1995) [inline clippy_lints/src/empty_enum.rs:13]: NIT:
```suggestion
    /// if the `never_type` feature is enabled.
```

File: clippy_lints/src/empty_enum.rs [hunk 2]
```diff
@@ -44,7 +48,9 @@ impl<'tcx> LateLintPass<'tcx> for EmptyEnum {
         if let ItemKind::Enum(..) = item.kind {
             let ty = cx.tcx.type_of(did);
             let adt = ty.ty_adt_def().expect("already checked whether this is an enum");
-            if adt.variants.is_empty() {
+
+            // Only suggest the never type if the feature is enabled
+            if adt.variants.is_empty() && cx.tcx.features().never_type {
```
Comment (flip1995) [inline clippy_lints/src/empty_enum.rs:53]: I think we should put this as the first check of the function, so:

```rust
if !cx.tcx.features().never_type {
    return;
}
```

in front of the `let did = ...` statement.

File: tests/ui/empty_enum.rs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 #![allow(dead_code)]
 #![warn(clippy::empty_enum)]
-
+// Enable never type to test empty enum lint
+#![feature(never_type)]
```
Comment (flip1995) [inline tests/ui/empty_enum.rs:4]: Also nice would be a test `empty_enum_without_never_type.rs` (or a better name) where the feature is not enabled and the `stderr` file is empty aka non-existent.

**Review by flip1995 — COMMENTED** [review]
Thanks! Just some NITS and a test missing.

**Comment by nahuakang** [comment]
@flip1995 Made changes according to your feedback. Thanks so much :pray:

**Comment by flip1995** [comment]
@bors r+

Thanks!

**Pull request merged**